### PR TITLE
Feat: added login and signup

### PR DESCRIPTION
**Pull request by bharati-21**
- Users can sign up with a new email.
- Users can login with guest credentials
- Users can login using their existing accounts

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,11 +3,24 @@ import "./styles/styles.css";
...
-import { Routes } from "./routes/Routes";
+import { Routes } from "./routes/WebsiteRoutes";
+
+/* External Imports */
+import { ToastContainer } from 'react-toastify';
 
 function App() {
     return (
         <div className="App">
+            <ToastContainer
```
Comment (sonishreyas) [inline src/App.js:14]: Really liked this.

File: src/routes/ProtectedRoutes.jsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { Navigate, Outlet } from 'react-router-dom';
+import { useAuth } from 'contexts/';
+
+const ProtectedRoutes = () => {
```
Comment (sonishreyas) [inline src/routes/ProtectedRoutes.jsx:4]: Really liked how you have used protectedRoutes

**Review by sonishreyas — APPROVED** [review]
The code looks good and easy to understand.

**Comment by bharati-21** [comment]
> The code looks good and easy to understand.

Thank you for your review!

**Review by yashpatel1797 — APPROVED** [review]
1. File structure is good. everything is divided into separate folders and files.
2. Good that use utils and created separate files for API calls.
3. Like that you have implemented useToast and showed the status of every action. I will try to add in my toastify as well.

**Comment by bharati-21** [comment]
> 1. File structure is good. everything is divided into separate folders and files.
> 2. Good that use utils and created separate files for API calls.
> 3. Like that you have implemented useToast and showed the status of every action. I will try to add in my toastify as well.

Thank you for your review!

**Pull request merged**